Add Community page that can be accessed from Forum

karenfrilya97 opened the pull request:
![image](https://user-images.githubusercontent.com/28861203/59624924-a1493080-916a-11e9-940d-31cc559a6209.png)

SuyashLakhotia commented on src/main/java/com/google/codeu/data/Datastore.java:
```diff
@@ -112,9 +112,9 @@ public void storeUser(User user) {
-  * Returns the User owned by the email address, or
-  * null if no matching User was found.
-  */
+   * Gets the User owned by the email address, or
```
Why this change? In Javadoc, we usually describe methods as "Returns ..." and not "Gets ...".

karenfrilya97 replied:
I followed the verb in the method name but sure I'll change to "Returns ..."

SuyashLakhotia commented on src/main/java/com/google/codeu/data/Datastore.java:
```diff
@@ -219,4 +225,75 @@ public void storeCommunity(Community community) {
+   * Gets a list of threads posted in a community, or empty list if the
+   * community has no thread. List is sorted by name ascending.
+   */
+  //TODO: find a better way to sort threads
```
Is there some sort of timestamp you could rely on?

karenfrilya97 replied:
I was thinking it should be sorted by popularity. But I'm not sure how to evaluate popularity of a Community. Some of the sorting methods I can think of are:
1. Sort by timestamp which reflects the last time a thread or comment was added to this Community.
2. Sort by the total number of comments across all the threads in a community.
3. Sort by the number of members in a community.

Not sure which is better. Or maybe can use a combination of a few or let the user choose the sorting method?

Anyway I was thinking of adding the sorting feature in a separate PR but do you think I should fix it before merging this PR?

SuyashLakhotia replied:
I like Option 1. Its the simplest. Maybe later we can have a sorting feature that allows the user to sort by a specific attribute. Feel free to do it in another PR.

karenfrilya97 replied:
Ok! I'll do it in another PR. I'll add it to the task tracker for now. Thank you!

SuyashLakhotia commented:
LGTM overall, just a couple minor changes requested.

If I understand correctly, each community has a list of threads and each thread has a tree of posts?

karenfrilya97 commented:
@SuyashLakhotia Yup. But I changed the name to comments instead of posts. 
So at first a thread has one root post which is the main post. But because datastore does not support user-defined classes, I just stored the main post as a text attribute of the thread. Then the contents of the thread more like replies or comments to this main post so I thought Comment is a better name than Post.

And I deleted Post.java from the Forum branch but I forgot to delete it here. I'm replacing it with the Comment.java in PR #32 .
